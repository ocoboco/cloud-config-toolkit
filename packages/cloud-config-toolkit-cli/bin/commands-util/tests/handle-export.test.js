const mock = require('mock-fs');
const fs = require('fs-extra');

const handleExport = require('../handle-export');

const toolkit = {
  deserialize: JSON.parse,
  serialize: JSON.stringify,
  validate() {
    return {
      isValid: true, 
      errors: []
    };
  },
  export(configuration) {
    return {
      ...configuration,
      exported: true
    };
  }
};

describe('handleExport()', function() {
  test('throws an error when destination file already exists', async function() {
    try {
      await handleExport(toolkit, {
        path: 'config.json',
        destination: 'exportedConfig.json'
      });
    } catch (e) {
      expect(e.message).toMatchSnapshot();
    }
  });

  test('throws an error when destination directory does not exist', async function() {
    try {
      await handleExport(toolkit, {
        path: 'config.json',
        destination: 'no-dir/exportedConfig.json'
      });
    } catch (e) {
      expect(e.message).toMatchSnapshot();
    }
  });

  test('writes exported config to destination file', async function() {
    await handleExport(toolkit, {
      path: 'config.json',
      destination: 'config-dir/exportedConfig.json'
    });
    const fileContent = await fs.readFile('config-dir/exportedConfig.json', 'utf8');
    expect(fileContent).toMatchSnapshot();
  });
});

beforeAll(function() {
  console.log = jest.fn();
  mock({
    'config.json': '{ "prop": "value" }',
    'exportedConfig.json': '{ "prop": "value", "exported": true }',
    'config-dir/': {}
  });
});

afterAll(function() {
  mock.restore();
});