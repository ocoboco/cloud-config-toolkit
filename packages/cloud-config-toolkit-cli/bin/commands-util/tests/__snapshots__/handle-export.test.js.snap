// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleExport() throws an error when destination directory does not exist 1`] = `"ENOENT, no such file or directory 'no-dir/exportedConfig.json'"`;

exports[`handleExport() throws an error when destination file already exists 1`] = `"Cannot write to destination file, a file already exists at 'exportedConfig.json'"`;

exports[`handleExport() writes exported config to destination file 1`] = `"{\\"prop\\":\\"value\\",\\"exported\\":true}"`;
